### PR TITLE
EVG-5972 Add option to force CLI update

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: operations/update.go [hunk 1]
```diff
@@ -47,88 +54,68 @@ func Update() cli.Command {
...
 				if err != nil {
 					return err
 				}
-				fmt.Println("Upgrade complete!")
+				grip.Info("Upgrade complete!")
 				return nil
 			}
 
-			fmt.Println("New binary downloaded (but not installed) to path: ", updatedBin)
+			grip.Infof("New binary downloaded (but not installed) to path: %s", updatedBin)
```
Comment (tychoish) [inline operations/update.go]: this is really infoln

File: operations/update.go [hunk 2]
```diff
@@ -47,88 +54,68 @@ func Update() cli.Command {
...
 
-				fmt.Println("Unlinking existing binary at", binaryDest)
+				grip.Infof("Unlinking existing binary at %s", binaryDest)
 				err = syscall.Unlink(binaryDest)
 				if err != nil {
 					return err
 				}
-				fmt.Println("Copying upgraded binary to: ", binaryDest)
-				err = copyFile(binaryDest, updatedBin)
+				grip.Infof("Moving upgraded binary to: %s", binaryDest)
```
Comment (tychoish) [inline operations/update.go]: infoln

File: operations/update.go [hunk 3]
```diff
@@ -47,88 +54,68 @@ func Update() cli.Command {
...
 
 			if doInstall {
-				fmt.Println("Upgraded binary successfully downloaded to temporary file:", updatedBin)
+				grip.Infof("Upgraded binary successfully downloaded to temporary file: %s", updatedBin)
 
 				var binaryDest string
 				binaryDest, err = osext.Executable()
 				if err != nil {
-					return errors.Errorf("Failed to get installation path: %v", err)
+					return errors.Errorf("Failed to get installation path: %s", err)
```
Comment (tychoish) [inline operations/update.go]: wrap err

File: operations/update.go [hunk 4]
```diff
@@ -183,14 +170,14 @@ func prepareUpdate(url, newVersion string) (string, error) {
 	cmd := exec.Command(tempPath, "version")
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		return "", errors.Errorf("Update failed - checking version of new binary returned error: %v", err)
+		return "", errors.Errorf("Update failed - checking version of new binary returned error: %s", err)
```
Comment (tychoish) [inline operations/update.go]: wraperr

File: operations/update.go [hunk 5]
```diff
@@ -212,24 +199,24 @@ func checkUpdate(client client.Communicator, silent bool) (updateStatus, error)
 	// server says is the latest
 	clients, err := client.GetClientConfig(context.Background())
 	if err != nil {
-		fmt.Fprintf(outLog, "Failed checking for updates: %v\n", err)
+		fmt.Fprintf(outLog, "Failed checking for updates: %s\n", err)
```
Comment (tychoish) [inline operations/update.go]: can't we do some grip.NoticeWhen(!silent, message.WrapError()) thing

File: operations/update.go (hunk 3 above)
Comment (bsamek) [reply operations/update.go]: done

File: operations/update.go (hunk 2 above)
Comment (bsamek) [reply operations/update.go]: done

File: operations/update.go (hunk 1 above)
Comment (bsamek) [reply operations/update.go]: done

File: operations/update.go (hunk 4 above)
Comment (bsamek) [reply operations/update.go]: I disagree. I don't think the stack is useful here. The error is about what shelling out to a binary returned.

File: operations/update.go (hunk 5 above)
Comment (bsamek) [reply operations/update.go]: We use this branch to return though.

File: operations/update.go (hunk 4 above)
Comment (tychoish) [reply operations/update.go]: I don't think we're printing the stack except in debugging/server side contexts. Wrapping just avoids weird string formatting

File: operations/update.go [hunk 6]
```diff
@@ -183,14 +170,14 @@ func prepareUpdate(url, newVersion string) (string, error) {
...
-		return "", errors.Errorf("Update failed - checking version of new binary returned error: %v", err)
+		return "", errors.Errorf("Update failed - checking version of new binary returned error: %s", err)
 	}
 
 	updatedVersion := string(out)
 	updatedVersion = strings.TrimSpace(updatedVersion)
 
 	if updatedVersion != newVersion {
-		return "", errors.Errorf("Update failed - expected new binary to have version %v, but got %v instead", newVersion, updatedVersion)
+		return "", errors.Errorf("Update failed - expected new binary to have version %s, but got %s instead", newVersion, updatedVersion)
```
Comment (tychoish) [inline operations/update.go]: wrap

File: operations/update.go (hunk 5 above)
Comment (bsamek) [reply operations/update.go]: kinda? I did a thing

File: operations/update.go [hunk 7]
```diff
@@ -170,7 +171,7 @@ func prepareUpdate(url, newVersion string) (string, error) {
 	cmd := exec.Command(tempPath, "version")
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		return "", errors.Errorf("Update failed - checking version of new binary returned error: %s", err)
+		return "", errors.Wrapf(err, "Update failed - checking version of new binary returned error: %s")
```
Comment (tychoish) [inline operations/update.go]: Wrap, omit %s

**Pull request merged**